### PR TITLE
 Prepare repository for next `.github` release and SBT build improvements

### PR DESCRIPTION
**Pull request by alejandrohdezma**
# What has been done in this PR?

## Prepare project for next `.github` release

- Add `ci-publish` command alias that will be used instead of `ci-release` for publishing artifacts.
- Move docs templates to `docs` folder (as mdoc is expected).
- Move microsite files to `microsite` folder.
- Run `github` before `ci-docs` & `ci-publish`.
- Remove `ci-microsite` (it gets inline in `ci-docs`).
- Add `actions.yml` file that will be used by next `.github` release workflows to read pre/post conditions to run before/after `ci-*` commands.

## Simplify SBT build

- Re-order settings.
- Remove unneeded settings.
- Remove old `sbt-org-policies` markers.
- Add [sbt-remove-test-from-pom](https://github.com/alejandrohdezma/sbt-remove-test-from-pom) to avoid publishing test dependencies to POM.
- Add [sbt-codecov](https://github.com/alejandrohdezma/sbt-codecov) to run tests with coverage and auto-upload to Codecov.

## Update versions

- Update plugins to latest versions
- Update Scala versions to `2.13.2`/`2.12.11`

### CODE REVIEW

File: project/ProjectPlugin.scala [hunk 1]
```diff
@@ -3,32 +3,25 @@ import microsites._
...
-  override def requires: Plugins = SbtGithubPlugin
-
   object autoImport {
 
     lazy val V = new {
-      val scala212: String   = "2.12.10"
-      val scala213: String   = "2.13.1"
-      val base64: String     = "0.2.9"
-      val cats: String       = "2.1.1"
-      val catsEffect: String = "2.1.1"
```
Comment (BenFradet) [inline project/ProjectPlugin.scala:25]: why was it removed?

File: project/ProjectPlugin.scala [hunk 2]
```diff
@@ -90,38 +81,26 @@ object ProjectPlugin extends AutoPlugin {
...
-      coverageExcludedPackages := "<empty>;github4s\\.scalaz\\..*",
-      // This is necessary to prevent packaging the BuildInfo with
-      // sensible information like the Github token. Do not remove.
-      mappings in (Compile, packageBin) ~= { (ms: Seq[(File, String)]) =>
-        ms filter {
-          case (_, toPath) =>
-            !toPath.startsWith("github4s/BuildInfo")
-        }
-      }
+      coverageExcludedPackages := "<empty>;github4s\\.scalaz\\..*"
```
Comment (BenFradet) [inline project/ProjectPlugin.scala:96]: unrelated to this PR but we can remove that 👍

File: project/ProjectPlugin.scala (hunk 1 above)
Comment (alejandrohdezma) [reply project/ProjectPlugin.scala:25]: It wasn't being used

File: project/ProjectPlugin.scala (hunk 2 above)
Comment (alejandrohdezma) [reply project/ProjectPlugin.scala]: Done in https://github.com/47degrees/github4s/pull/501/commits/2d14932947cf1f51fc6bee242e202e41ecbe211d! Thanks!

File: project/ProjectPlugin.scala (hunk 1 above)
Comment (BenFradet) [reply project/ProjectPlugin.scala:25]: Thanks, logged #502

**Pull request merged**